### DISCUSSION TITLE
Problem with compiler version in VSCode

### DISCUSSION THREAD

**Discussion by 0xh0551** [Cyfrin/foundry-full-course-cu#569] [Q&A] [answered]
Hello all. Hope you've started a great day.

I'm coming along with Patrick. Now, I stuck in lesson 6, where VSCode and foundry is coming to use.
I installed VSCode and needed extensions, installed foundry (using foundryup). Used "forge -init", deleted all unnecessary files as per Patrick's guidance.
Then copied the SimpleStorage.sol to the VSCode, src directory. VS code shows me an error for the pragma line. 
```solidity
pragma solidity ^0.8.19;
```

Which comes with 3 strange matters:

1. I got error on compiler version:
```solidity
Source file requires different compiler version (current compiler is 0.4.17+commit.bdeb9e52.Emscripten.clang - note that nightly builds are considered to be strictly less than the released version
```
although, I chaned the compiler version using Solidity extension by Juan Blanco to latest, still got the same error (current compiler is 0.4.17)

2. I changed compiler version to 0.4.17 to see whether the error vanished or not. Surprisingly the error has changed as below:
```solidity
Source file requires different compiler version (current compiler is 0.8.19+commit.7dd6d404.Emscripten.clang) - note that nightly builds are considered to be strictly less than the released version
```
now it says the compiler version is 0.8.19!!!

3. I used the command "forge compile". it immediately goes to download solc 0.8.21 then pass timeout error as below:
<img width="633" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/71399850/b0958f91-9fd4-404b-ab3f-1fdcdbd652e5">

Now I stuck here and can not learn further :D
Any help would be appreciated.

**Comment by shikhar229169** [comment]
Can you restart your vs code and try again.
Sometimes there are network issues and they get fixed on restarting vs code.

**Reply by 0xh0551** [reply]
Uhh shikhar :( I did it couple of times 
even I restarted my mac :-/
which by the way did not work :-/

**Reply by shikhar229169** [reply]
Can you try to run the below command:
```
ping google.com
```

**Reply by 0xh0551** [reply]
dude! I have and stable connection. (which I'm using to reply your comment here:D)
but if you are looking for sth else, here is the result:
(changed the ip to "myip")
```
64 bytes from myip : icmp_seq=0 ttl=111 time=168.759 ms
64 bytes from myip: icmp_seq=1 ttl=111 time=282.877 ms
64 bytes from myip: icmp_seq=2 ttl=111 time=177.355 ms
64 bytes from myip: icmp_seq=3 ttl=111 time=245.901 ms
64 bytes from myip: icmp_seq=4 ttl=111 time=251.155 ms
64 bytes from myip: icmp_seq=5 ttl=111 time=64.987 ms
64 bytes from myip: icmp_seq=6 ttl=111 time=361.213 ms
64 bytes from myip: icmp_seq=7 ttl=111 time=301.823 ms
64 bytes from myip: icmp_seq=8 ttl=111 time=281.058 ms
64 bytes from myip: icmp_seq=9 ttl=111 time=234.030 ms
64 bytes from myip: icmp_seq=10 ttl=111 time=224.693 ms
```

**Reply by shikhar229169** [reply]
Yah but sometimes due to glitches there is internet problem in linux, my friend faced it several times, so I thought maybe this was the same case.

Can you ask ChatGpt about it, I can't get any idea about it.

**Reply by 0xh0551** [reply]
I used Phind and ChatGPT will send the results as a new comment.
still problem exist :-/

**Comment by 0xh0551** [comment]
I really stuck here and did everything I think might help. like uninstalling npm, brew, restart, then install brew again and then installed solidity. agian problem exist.

I used Phind, it gave me 3 possible solutions which did not work BTW.
but there was two comments which may be important to know:

1. Even if solc is installed elsewhere on your system and available in your PATH, forge may not be able to use it if it's not installed in the ~/.svm directory. This could explain why solc appears to be installed correctly when you run solc --version, but forge build is unable to find it.

I think it is right. (below terminal screenshot says it all:D ```forge build``` says I should install solc 0.8.21 but ```solc --version``` gave me the version 0.8.21) The only way is to install solc in ~/.svm directory by forge. (the versions I copied in ~/.svm did not work :-/
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/71399850/ef8d0e71-d0bb-4aaf-8ad9-163beaa6a5b2)


---
2.If none of these solutions work, there may be a bug in forge or svm causing this problem. In that case, I would recommend reporting the issue on the Foundry GitHub repository.
Is it a bug?

Help!

**Comment by theirrationalone** [comment]
Hey,

Idk, Why it is set to **0.4.17** (by default | You messed up something)??

Do You know that foundry has full support for **pragmatic-information** / (compiler), to play with it.

Now try this 👇...

```toml
solc_version = "version number inside double quotes"
```

wishing this works for you.😉

**Reply by 0xh0551** [reply]
thank you irrational

tried but not helped :(

**Comment by 0xh0551** [comment]
I think the problem is with svm. (Solidity Version Manager)
I can not install it because of a 404 error 😁

the address of SVM is broken or something.

```terminal
hosseinnarimani@Hosseins-MacBook-Air SimpleStorage % curl -L https://github.com/web3j/svm/raw/master/install.sh | bash && source ~/.svm/svm.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  2648  100  2648    0     0   2633      0  0:00:01  0:00:01 --:--:-- 1778k
Downloading svm...
https://github.com/web3j/svm/releases/download/0.1.2/svm-darwin.tar
Looks like there was an error while trying to download svm
source: no such file or directory: /Users/hosseinnarimani/.svm/svm.sh
```

the webaddress ```https://github.com/web3j/svm/releases/download/0.1.2/svm-darwin.tar``` goes to nowhere :-/

**Reply by theirrationalone** [reply]
Repo/Project not exist.

**Reply by 0xh0551** [reply]
yess and where should I download this Version Manager? :(

**Reply by theirrationalone** [reply]
Try this... 51% chances, tou will get resolved... 😅😅

[Rust Repo](https://rustrepo.com/repo/roynalnaruto-svm-rs-rust-miscellaneous)

**Comment by 0xh0551** [comment, marked as answer]
mind is blowing!!

changed DNS from 8.8.8.8 to 1.1.1.1 and problem solved :[
